MantidQt/CustomInterfaces/src/Tomography/ImggFormatsConvertPresenter.cpp

```diff
@@ -0,0 +1,315 @@
+ * @param outputName name of the output image to produce
+ * @param outFormat format for the output image
+ */
+void ImggFormatsConvertPresenter::convert(const std::string &inputName,
+                                          const std::string &inFormat,
+                                          const std::string &outputName,
+                                          const std::string &outFormat) const {
+
+  if ("NXTomo" == outFormat) {
+    convertToNXTomo(inputName, inFormat, outputName);
+  } else if ("FITS" == outFormat) {
+    convertToFITS(inputName, inFormat, outputName);
+  } else {
+    if ("FITS" == inFormat) {
+      auto inWks = loadFITS(inputName);
+      m_view->writeImg(inWks, outputName, outFormat);
+    } else {
+      // other image formats
+      m_view->convert(inputName, inFormat, outputName, outFormat);
+    }
+  }
+}
+
+void ImggFormatsConvertPresenter::convertToFITS(
+    const std::string &inputName, const std::string &inFormat,
+    const std::string &outputName) const {
+
+  Mantid::API::MatrixWorkspace_sptr inWks;
+  if ("FITS" == inFormat) {
+    inWks = loadFITS(inputName);
+  } else {
+    inWks = m_view->loadImg(inputName, inFormat);
+  }
+
+  saveFITS(inWks, outputName);
+}
+
+void ImggFormatsConvertPresenter::convertToNXTomo(
+    const std::string &inputName, const std::string &inFormat,
+    const std::string &outputName) const {
+  // loadNXTomo is not enabled (and there is no LoadNXTomo algorithm for now)
+
+  auto inWks = m_view->loadImg(inputName, inFormat);
+  saveNXTomo(inWks, outputName);
+}
+
+Mantid::API::MatrixWorkspace_sptr
+ImggFormatsConvertPresenter::loadFITS(const std::string &inputName) const {
+
+  const std::string wksGrpName = "__fits_img_to_convert";
+  if (Mantid::API::AnalysisDataService::Instance().doesExist(wksGrpName)) {
+    auto algDel =
+        Mantid::API::AlgorithmManager::Instance().create("DeleteWorkspace");
+    algDel->setChild(true);
+    algDel->setPropertyValue("Workspace", wksGrpName);
+    algDel->execute();
+  }
+  // Just run LoadFITS
+  auto alg = Mantid::API::AlgorithmManager::Instance().create("LoadFITS");
+  alg->initialize();
+  alg->setProperty("Filename", inputName);
+  alg->setProperty("OutputWorkspace", wksGrpName);
+  alg->setProperty("LoadAsRectImg", true);
+  alg->execute();
+
+  if (!alg->isExecuted()) {
+    throw std::runtime_error("Failed to execute the algorithm "
+                             "LoadFITS to load the image file '" +
+                             inputName + "' in FITS format.");
+  }
+
+  // What should be alg->getProperty("OutputWorkspace"); but this is a group
+  Mantid::API::WorkspaceGroup_sptr grp =
+      Mantid::API::AnalysisDataService::Instance()
+          .retrieveWS<Mantid::API::WorkspaceGroup>(wksGrpName);
+  Mantid::API::MatrixWorkspace_sptr imgWorkspace =
+      boost::dynamic_pointer_cast<Mantid::API::MatrixWorkspace>(
+          grp->getItem(grp->size() - 1));
+
+  return imgWorkspace;
```

Comment:
Now that you mention this I'm going to double check. I think here it should be possible to do it the neat way with `getProperty()`.